packages/pds/src/label-cache.ts

```diff
@@ -0,0 +1,90 @@
+import { createDeferrable, wait } from '@atproto/common'
+import Database from './db'
+import { Label } from './db/tables/label'
+
+export class LabelCache {
+  bySubject: Record<string, Label[]> = {}
+  latestLabel = ''
+  defer = createDeferrable()
+  refreshes = 0
+
+  destroyed = false
+
+  constructor(public db: Database) {}
+
+  async start() {
+    await this.fullRefresh()
+    this.poll()
```

Comment:
This seems unlikely to throw, but for good measure it might make sense to add a `catch()` and log to avoid an unhandled rejection.

Reply:
sweet, I ended up just wrapping the execution of `poll` in a try/catch similar to what we do in the sequencer